core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1071,17 +1095,12 @@ public void run() {
 
     private <ReqT, RespT> ClientCall<ReqT, RespT> newClientCall(
         MethodDescriptor<ReqT, RespT> method, CallOptions callOptions) {
-      return new ClientCallImpl<>(
-          method,
-          getCallExecutor(callOptions),
-          callOptions,
-          transportProvider,
-          terminated ? null : transportFactory.getScheduledExecutorService(),
-          channelCallTracer,
-          configSelector.get())
-          .setFullStreamDecompression(fullStreamDecompression)
-          .setDecompressorRegistry(decompressorRegistry)
-          .setCompressorRegistry(compressorRegistry);
+      InternalConfigSelector selector = configSelector.get();
+      if (selector == null) {
```

Comment:
We need a fast path for when the selector is set, but it is a ManagedChannelServiceConfig.getDefaultConfigSelector(). We should only pay the indirection cost when it is necessary.

Reply:
Added a fast path for this below.